### ISSUE TITLE
obspy.read() unaffected by byteorder settings for SAC files

### ISSUE THREAD

**Issue by guyomd** [obspy/obspy#2285]
I am under the impression that obspy.read() behavior seems unaffected by setting 'byteorder' settings.
For example, I was not able to load a little-endian SAC trace (created by an home-made Fortran routine of my own). Even if I tried specifying explicitely "byteorder=little" in option.
However, reformatting the same record using a big-endian byteorder using the same routine produced a file that I could load properly using obspy.read() with no option.
Note: My computer is natively ordering bytes into little-endian representation.

My conclusion is that osbpy.read() can only handle big-endian files for SAC format (independently of  byteorder option settings)

Both files are provided in attachement (same record, produced by the same routine, only byteordering differs).

[endian_read_issue_obspy.zip](https://github.com/obspy/obspy/files/2773260/endian_read_issue_obspy.zip)

**Comment by megies** [comment]
Yep, `_internal_read_sac()` is not passing through all `kwargs` to `SACTrace.read()` properly.

As a workaround, you could do this in principle:

```python
from obspy.io.sac.sactrace import SACTrace
with open('SIS.DANS.SHE.20030213_193452_LITTLE_ENDIAN.sac', 'rb') as fh:
    sac = SACTrace.read(fh, ascii=False, checksize=True, encoding='ascii', byteorder='little')
    tr = sac.to_obspy_trace(encoding='ascii')
```

However this fails because `nvhdr` field has an unexpected value because it is unset (`-12345`) and that's expected to be a value from 0 to 20 apparently. So `sac.arrayio.is_valid_byteorder()` should be changed as well, since `-12345` may potentially encountered in any header field I guess. 

So, actually, you need some monkey patching to read your file:

```python
import mock
from obspy.io.sac.sactrace import SACTrace
with mock.patch('obspy.io.sac.arrayio.is_valid_byteorder') as p:
    p.return_value = True
    with open('SIS.DANS.SHE.20030213_193452_LITTLE_ENDIAN.sac', 'rb') as fh:
        sac = SACTrace.read(fh, ascii=False, checksize=True, encoding='ascii', byteorder='little')
        tr = sac.to_obspy_trace(encoding='ascii')
```


CC @jkmacc-LANL

**Comment by jkmacc-LANL** [comment]
Thanks for reporting this @guyomd and looking into it @megies !  I'll try to fix this quickly.

**Comment by jkmacc-LANL** [comment]
@guyomd Can the files written by your Fortran routine be read by the [SAC program](https://ds.iris.edu/ds/nodes/dmc/software/downloads/sac/)?  I ask because I'm hesitant to allow the `nvhdr` to go unset if that means it passes files that wouldn't be readable by the canonical SAC program.

**Comment by guyomd** [comment]
I cannot say ... I have not tried using the SAC program. ..



De: notifications@github.com
Envoyé: 24 janvier 2019 4:54 AM
À: obspy@noreply.github.com
Répondre à: reply@reply.github.com
Cc: guillaume.daniel@outlook.com; mention@noreply.github.com
Objet: Re: [obspy/obspy] obspy.read() unaffected by byteorder settings for SAC files (#2285)



@guyomd<https://github.com/guyomd> Can the files written by your Fortran routine be read by the SAC program<https://ds.iris.edu/ds/nodes/dmc/software/downloads/sac/>? I ask because I'm hesitant to allow the nvhdr to go unset if that means it passes files that wouldn't be readable by the canonical SAC program.

**Comment by megies** [comment]
fixed by #2292
